src/docfx/lib/markdown/ExtractTitle.cs

```diff
@@ -26,6 +28,35 @@ public static MarkdownPipelineBuilder UseExtractTitle(this MarkdownPipelineBuild
             });
         }
 
+        private static HeadingBlock GetFirstAfterYamlNoneCommentHeadingBlock(MarkdownDocument document)
+        {
+            var firstAfterYamlBlock = GetAfterYamlChildren(document).SkipWhile(block => IsCommentsBlock(block)).FirstOrDefault();
+            return firstAfterYamlBlock as HeadingBlock;
+        }
+
+        private static IEnumerable<Block> GetAfterYamlChildren(MarkdownDocument document)
+        {
+            return document.FirstOrDefault() is YamlFrontMatterBlock ? document.Skip(1) : document;
+        }
+
+        private static bool IsCommentsBlock(Block block)
+        {
+            var htmlBlock = block as HtmlBlock;
+
+            if (htmlBlock != null)
+            {
+                var document = new HtmlDocument();
```

Comment:
instead of using HtmlAgilityPack, can we quickly detect comment block using starts with `<!--` and ends with `-->` ?

Reply:
use HtmlBlock.Type instead

---
In reply to: [210175472](https://github.com/dotnet/docfx/pull/3192#discussion_r210175472) [](ancestors = 210175472)